cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -0,0 +1,229 @@
+      if (locationFactory != null && this.locationManager == null) {
+        this.locationManager = new LocationManager(locationFactory);
+        started = true;
+      }
+
+      // logic from AppenderBase
+      if (!this.started) {
+        addStatus(new WarnStatus(
+          "Attempted to append to non started appender [" + name + "].",
+          this));
+        return;
+      }
+
+      // logic from AppenderBase
+      if (getFilterChainDecision(eventObject) == FilterReply.DENY) {
+        return;
+      }
+
+      String namespaceId = eventObject.getMDCPropertyMap().get(LocationManager.TAG_NAMESPACE_ID);
+      if (!namespaceId.equals(NamespaceId.SYSTEM.getEntityName())) {
+        LocationIdentifier logLocationIdentifier = locationManager.getLocationIdentifier(eventObject
+                                                                                           .getMDCPropertyMap());
+        rollover(logLocationIdentifier, eventObject);
+        OutputStream locationOutputStream = locationManager.getLocationOutputStream(logLocationIdentifier, fileName);
+        setOutputStream(locationOutputStream);
+        writeOut(eventObject);
+      }
+    } catch (IllegalArgumentException iae) {
+      // this shouldn't happen
+      LOG.error("Unrecognized context ", iae);
+    } catch (IOException ioe) {
+      throw new LogbackException("Exception during append", ioe);
+    }
+  }
+
+  private void rollover(LocationIdentifier logLocationIdentifier, ILoggingEvent eventObject) {
+    if (!locationManager.getActiveFilesLocations().containsKey(logLocationIdentifier)) {
+      return;
+    }
+
+    Location logFile = locationManager.getActiveFilesLocations().get(logLocationIdentifier);
+    triggeringPolicy.setLocation(logFile);
+
+    // Since TriggeringPolicy#isTriggeringEvent method takes File in the api, we provide dummy file to our
+    // triggeringPolicy implementation
+    if (triggeringPolicy.isTriggeringEvent(new File("dummy"), eventObject)) {
+      try {
+        rollingPolicy.setLocation(locationManager.getLogLocation(logLocationIdentifier), logFile);
+        rollingPolicy.rollover();
+
+        // clear caches so new location gets created
+        locationManager.closeOutputStream(locationManager.getActiveFiles().get(logLocationIdentifier));
+        locationManager.getActiveFiles().remove(logLocationIdentifier);
+        locationManager.getActiveFilesLocations().remove(logLocationIdentifier);
+
+      } catch (IOException e) {
+        throw new LogbackException("Exception while rolling over", e);
+      } catch (RolloverFailure e) {
+        // we do not want to stop processing because roll over failed so catch it and process the event
+        addStatus(new WarnStatus("Attempt to rollover failed for appender [" + name + "].", this));
+        LOG.info("Attempt to rollover failed fro appender [" + name + "].", this);
+      }
+    }
+  }
+
+  @Override
+  public void flush() throws IOException {
+    // no-op
```

Comment:
hmm. For this, I have overridden `OutputStreamAppender#setLayout` and did `immediateFlush = false`. This means, wrapped encoder used to append log events will no longer be flushing on every log event. Instead, flush can be applied when `Appender#flush()` is called.